contents/security/passwordStorage.md

```diff
@@ -0,0 +1,113 @@
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Password Storage
+
+**Author: [Jeremy Choo](https://github.com/ChooJeremy)** <br>
+Reviewers: [Amrut Prabhu](https://github.com/amrut-prabhu), [Marvin Chin](https://github.com/marvinchin), [Tan Zhen Yong](https://github.com/Xenonym), [Wang Junming](https://github.com/junming403)
+
+## Overview
+
+Many software applications use a username and password combination as user account credentials for authentication. Obviously, it is not a good idea for the software to store these credentials as <trigger for="pop:plaintext">plain text</trigger> because if someone else were to gain access to them either lawfully (e.g., an employee who has access to the data) or unlawfully (e.g., someone hacking into the data storage), that person can use those credentials directly to impersonate the account owner. This article explains some techniques that can be used to store user credentials more securely:
+
+- Encryption
+- Hashing
+- Salting
+
+<popover id="pop:plaintext" title="_Plaintext_ refers to unencrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to text before encyption or after decrypting it. Another term for it is cleartext.
+  </div>
+</popover>
+
+<popover id="pop:oneway" title="" placement="top">
+<div slot="content">
+	A one way function is a function that is easy to compute the result of, but whose results are difficult to reverse back to the original input.
+</div>
+</popover>
+
+## What is encryption?
+
+Encryption refers to the process of changing a message such that it becomes essentially random text. The opposite process is called decryption, which converts the text back into the original message. Encryption might seem like a good idea because it transfoms the data into essentially random bytes, which prevents all of the problems with storing the data directly in plaintext. However, because encryption is <tooltip content="If a function is reversible, and it converts from x to y, then it can also convert from y back to x"> reversible</tooltip>, it is always possible to regain the original password from the <trigger for="pop:ciphertext">ciphertext</trigger>.
+
+<popover id="pop:ciphertext" title="_Ciphertext_ refers to encrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to data after encrypting it.
+  </div>
+</popover>
+
+While encryption might seem suitable for storing passwords, it faces some severe problems. Since the data is encrypted, the encryption key must also be stored somewhere. This means that if someone manages to hack into the application and read the encrypted passwords, it is also likely that they will be able to read the encryptio key. With the encryption key, they will be able to decrypt all the passwords and read them anyway. This makes encryption unsuitable for password storage.
+
```

Comment:
This section seems to say everything twice. Can try to explain in one pass?
* encryption turns data into cyphertext by combining it with a key
* decryption can get the original data from cypertext if the key is available
* hence encryption is not suitable for secure storage of passwords